### DISCUSSION TITLE
Markdown parsing has changed dramatically

### DISCUSSION THREAD

**Discussion by dimaip** [mdx-js/mdx#2158] [Q&A] [answered]
Hi!
I've upgraded "@mdx-js/loader": from "^2.0.0-next" to "^2.1.5", and now Markdown parsing has started behaving completely differently, e.g. all of these used to be parsed as bold, but are not any more:

`**aaaa **`
`aaaa**)**aaaa`

Is there any way to bring old parsing back? I have tons of content and it's impossible to fix it for me....

Here's full package.json if it helps: https://github.com/dimaip/calendar/blob/master/package.json
Webpack: https://github.com/dimaip/calendar/blob/master/webpack.base.js#L132

**Comment by wooorm** [comment]
That’s how CommonMark, which is used in lots of places, including here on GitHub, works.
MDX 2 follows the specification. It’s not possible to divert.

**Comment by ChristianMurphy** [comment, marked as answer]
Hey @dimaip!
Sorry to hear you ran into a spot of trouble, happy to talk through options.

> now Markdown parsing has started behaving completely differently

To be clear, markdown parsing stayed largely the same.
The only differences are some bugfixes to correctly align edge cases to the CommonMark and GFM standards.

> all of these used to be parsed as bold, but are not any more

These are some of the edge cases.
Previewing that code in GitHub itself you can see it is not bolded either.

---

**aaaa **

aaaa**)**aaaa

---

> Is there any way to bring old parsing back?

No, bugs will not be re-added to MDX.
And especially not bugs from a pre-alpha release.

> I have tons of content and it's impossible to fix it for me....

I'd recommend transforming content with code to make it valid CommonMark/GFM in an automated way.
https://unifiedjs.com/learn/ introduces abstract syntax trees and transformers, I'd suggest familiarizing yourself with these, if you aren't already.
And create transformers to correct/adjust content.

**Reply by dimaip** [reply]
Thanks for your feedback, folks!
I was afraid to hear that :-) I understand that it's the "correct" behaviour (though the original behaviour was far more sensible).

Could you share any link to those bugfixes? For now, I'll try to at least revert to the last release before the parsing behaviour has changed, while trying to fix the content...

And speaking about the case of `aaaa**)**aaaa`, what's actually the correct way to write this in CommonMark? I only know of `aaa<b>)</b>aaa`...

**Reply by ChristianMurphy** [reply]
> Could you share any link to those bugfixes? 

Most likely https://github.com/mdx-js/mdx/pull/1367 or https://github.com/mdx-js/mdx/pull/1039
Though it's hard to pinpoint because I don't know what versions you are testing.
You could bisected the version which changed locally to get a better idea.

Reading between the lines, it sounds like you want to patch the parser to re add the bug?
I'd recommend against that for a couple reasons.
1. The change is most likely a couple levels down in the dependency tree.
2. Keeping invalid content and patching the bug back in will likely require re-patching for each release

> And speaking about the case of `aaaa**)**aaaa`, what's actually the correct way to write this in CommonMark?

A few options:
1. Use jsx syntax
2. Add zero width spaces around the part you want to be bolded

**Reply by ChristianMurphy** [reply]
Adding on a bit more for representing emphasis and strong text.
This write up https://github.com/orgs/syntax-tree/discussions/60#discussioncomment-2111096 shows ways to handle a number of edge cases.

**Reply by dimaip** [reply]
My apologies, forgot to respond here. I did end up regexp'ing a bunch of those mistakes, I think we mostly got it to work now.
Thanks so much for the links, they really did help.
